src/core/harpoon.ml

```diff
@@ -79,8 +73,14 @@ end
  *
  * Tactics are not obligated to solve the current subgoal!
  *)
-module Tactic (T : TacticContext) = struct
-  type t = unit Comp.proof_state -> unit
+module Tactic = struct
+  type t = unit Comp.proof_state -> tactic_context -> unit
+
+  and tactic_context =
```

Comment:
Since `tactic_context` doesn't depend on the definition of `t`, you should just define `tactic_context` before defining `t` and use `type` for both.